core/templates/dev/head/editor/ExplorationHistory.js

```diff
@@ -48,20 +49,61 @@ oppia.controller('ExplorationHistory', [
   var nodesData = null;
 
   $scope.$on('refreshVersionHistory', function(evt, data) {
+      $scope.clearAllCheckboxes();
     if (data.forceRefresh || $scope.explorationVersionMetadata === null) {
       $scope.refreshVersionHistory();
     }
   });
 
+  // Uncheck all checkboxes when page is refreshed
+  $scope.clearAllCheckboxes = function() {
+    angular.forEach($scope.snapshotOrderArray, function(versionCheckbox) {
+      versionCheckbox.selected = false;
+    }); 
+  };
+  
+  // Compares the two selected versions and displays the comparison results.
+  $scope.compareSelectedVersions = function() {
+    if($scope.selectedVersionsArray.length === 2) {
+      $scope.changeCompareVersion();
+      $scope.showHistoryGraph();
+    }
+  };
+  // Changes the checkbox selection and provides an appropriate user prompt.
+  $scope.changeSelectedVersions = function($event, versionNumber) {
```

Comment:
Let's use 'evt' as the first variable (this is consistent with other parts of the codebase which do similar things).